sklearn/neighbors/_base.py

```diff
@@ -79,7 +79,10 @@
 VALID_METRICS_SPARSE = dict(
     ball_tree=[],
     kd_tree=[],
-    brute=(PAIRWISE_DISTANCE_FUNCTIONS.keys() - {"haversine", "nan_euclidean"}),
+    brute=(
+        PAIRWISE_DISTANCE_FUNCTIONS.keys()
+        - {"haversine", "nan_euclidean", "nan_manhattan"}
```

Comment:
By importing `_NAN_METRICS` from `sklearn.metrics` if there aren't any circular imports, I think this removes redundancy.
```suggestion
        - {"haversine", **_NAN_METRICS}
```